homeworks/simonlaszlo/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,55 @@
+
+user_info={
+    "name":"Mike",
+    "age": 25,
+    "favourite_meals": [
+        "pizza",
+        "carbonara",
+        "sushi"
+    ],
+    "phone_contacts": {
+        "Mary": "+36701234567",
+        "Tim": "+36207654321",
+        "Tim2": "+36304567321",
+        "Jim": "+364005000"
+    }
+}
+skills_in = input("Kerek 4 programozasi nyelvet, vesszővel elválasztva, szóközök nélkül: ") # 1. adatbekeres
+user_info["skills"]=(skills_in.split(","))
+#print(user_info["skills"])
+
+user_info["favourite_meals"].sort() # 2. sorbarendezes
+#print(user_info["favourite_meals"])
+
+print(user_info["favourite_meals"]) # 3. az utolso elotti elem
+print(user_info["favourite_meals"][-2])
+
+user_info["favourite_meals"].append("spaghetti") # 4. elem hozzaadasa
+#print(user_info["favourite_meals"])
+
+user_info["favourite_meals"].extend(user_info["favourite_meals"][2:4]) # 5. ket elem hozzaadasa
+#print(user_info["favourite_meals"])
+
+user_info["favourite_meals"] = list(dict.fromkeys(user_info["favourite_meals"])) # 6. duplikacio torles
+#print(user_info["favourite_meals"])
+
+user_info["favourite_meals"].append(user_info["favourite_meals"][0]) # 7. elemek felcserelese
```

Comment:
ez így nem a leghatékonyabb megoldás, próbáld meg ezt: (így egy soron tudod végrehajtani a konkrét cserét az index-ek között)

`user_info["favourite_meals"][0],user_info["favourite_meals"][-1]=user_info["favourite_meals"][-1],user_info["favourite_meals"][0]`

Reply:
Erről beszélhetnénk órán? Elég formabontó számomra.

Reply:
Ez a tuple unpacking-gel kapcsolatos, a 4. órán tanultunk róla (06_unpacking.py).
Itt a háttérben az történik hogy az egyenlőség jel oldalán a vessző miatt létrejön egy tuple a jelenlegi -1. és 0. elemmel, és a bal oldali kifejezés az unpacking, amikor is a 0. elemnek adjuk a -1. elem értékét, és a -1. elemnek a 0. elem értékét.